backend/corpora/common/utils/db_utils.py

```diff
@@ -54,13 +60,19 @@ def commit(self):
             try:
                 self.session.commit()
             except SQLAlchemyError:
-                self.session.rollback()
-                msg = "Failed to commit."
-                logger.exception(msg)
-                raise CorporaException(msg)
+                self.failure_handler()
+
+        def failure_handler(self):
```

Comment:
Maybe you could do this in a context manager instead of repeating the try except for all of the functions?
Something like
```
@contextmanager
def session_scope():
    """Provide a transactional scope around a series of operations."""
    session = Session()
    try:
        yield session
        session.commit()
    except:
        session.rollback()
        raise
    finally:
        session.close

def run_my_program():
    with session_scope() as session:
        ThingOne().go(session)
        ThingTwo().go(session)
```
From https://docs.sqlalchemy.org/en/13/orm/session_basics.html#when-do-i-construct-a-session-when-do-i-commit-it-and-when-do-i-close-it

Reply:
This is implemented here:
https://github.com/chanzuckerberg/corpora-data-portal/blob/24d6cbe6ba28a8126dd9fdf1ad1647bbb0bf16cf/backend/corpora/common/utils/db_utils.py#L77
but only per API call:
https://github.com/chanzuckerberg/corpora-data-portal/blob/24d6cbe6ba28a8126dd9fdf1ad1647bbb0bf16cf/backend/corpora/lambdas/api/v1/dataset.py#L8

I added a test teardown to the failing tests to make sure the sessions are cleaned up properly.